src/perception/cost_map/cost_map.processing.cpp

```diff
@@ -0,0 +1,54 @@
+#include "cost_map.hpp"
+
+namespace mrover {
+
+    auto CostMapNodelet::pointCloudCallback(sensor_msgs::PointCloud2ConstPtr const& msg) -> void {
+        assert(msg);
+        assert(msg->height > 0);
+        assert(msg->width > 0);
+
+        if (!mPublishCostMap) return;
+
+        try {
+            SE3d zed_to_map = SE3Conversions::fromTfTree(mTfBuffer, "map", "zed_left_camera_frame");
+            auto* points = reinterpret_cast<Point const*>(msg->data.data());
+            // std::for_each(points, points + msg->width * msg->height, [&](auto* point) {
```

Comment:
I'm pretty sure this iteration method bands vertically. What I mean is that it skips more horizontally rather than vertically. A good solution would be to make a custom iterator that is cognizant of the current row so it knows to skip them. Then we can use that iterator in std::for_each with par_unseq.